lib/datadog/appsec/contrib/rack/gateway/response.rb

```diff
@@ -19,9 +19,58 @@ def initialize(body, status, headers, scope:)
               @scope = scope
             end
 
+            def parsed_body
+              return unless Datadog.configuration.appsec.parse_response_body
+
+              unless body.instance_of?(Array) || body.instance_of?(::Rack::BodyProxy)
+                if result.timeout
+                  Datadog.logger.debug do
+                    "Unable to parse response body because of unsupported body type: #{body.class}"
+                  end
+                end
+                return
+              end
+              return unless supported_response_type
+
+              body_dup = body.dup # avoid interating over the body. This is just in case code.
+              result = ''.dup
+              all_body_parts_are_string = true
+
+              body_dup.each do |body_part|
+                if body_part.is_a?(String)
+                  result.concat(body_part)
+                else
+                  all_body_parts_are_string = false
+                  break
+                end
+              end
+
+              return unless all_body_parts_are_string
+
+              if json?
+                JSON.parse(result)
+              else
+                result
+              end
+            end
+
             def response
               @response ||= ::Rack::Response.new(body, status, headers)
             end
+
+            private
+
+            def supported_response_type
+              json? || text?
+            end
+
+            def json?
+              headers['content-type'].include?('json')
+            end
+
+            def text?
+              headers['content-type'].include?('text')
```

Comment:
I'd rather be strict and explicitly list supported content types.

Indeed as is this would match at least `text/plain` and `text/html` but is there any point of passing those raw to libddwaf in the non-raw address that aims to carry parsed, structured data?

Is it for `text/xml`, which according to [RFC 3023](https://www.rfc-editor.org/rfc/rfc3023) says:

> If an XML document -- that is, the unprocessed, source XML document
   -- is readable by casual users, text/xml is preferable to
   application/xml.  MIME user agents (and web user agents) that do not
   have explicit support for text/xml will treat it as text/plain, for
   example, by displaying the XML MIME entity as plain text.
   Application/xml is preferable when the XML MIME entity is unreadable
   by casual users.

Reply:
You are right, since we do not parse, there is no schema information to extract from it 

I will remove the entire support for `text/*` content-type